erasure-readfile_test.go

```diff
@@ -0,0 +1,148 @@
+	if err != nil {
+		t.Errorf("getReadDisks returned error: %s", err)
+		return
+	}
+
+	for index := 0; index < len(readDisks); index++ {
+		if index == 0 {
+			if readDisks[index] != nil {
+				t.Errorf("readDisks[%d] should be nil", index)
+			}
+		}
+		if index < xl.dataBlocks {
+			if readDisks[index] != disks[index] {
+				t.Errorf("readDisks[%d] is incorrect", index)
+				return
+			}
+		} else {
+			if index < xl.dataBlocks+2 {
+				if readDisks[index] != disks[index] {
+					t.Errorf("readDisks[%d] is incorrect", index)
+					return
+				}
+			} else {
+				if readDisks[index] != nil {
+					t.Errorf("readDisks[%d] should be nil", index)
+				}
+			}
+		}
+	}
+}
+
+// Test getOrderedDisks.
+func testGetOrderedDisks(t *testing.T, xl xlObjects) {
+	disks := xl.storageDisks
+	blockCheckSums := make([]checkSumInfo, len(disks))
+	distribution := []int{16, 14, 12, 10, 8, 6, 4, 2, 1, 3, 5, 7, 9, 11, 13, 15}
+	orderedDisks, _ := getOrderedDisks(distribution, disks, blockCheckSums)
+	if orderedDisks[0] != disks[8] ||
+		orderedDisks[1] != disks[7] ||
+		orderedDisks[2] != disks[9] ||
+		orderedDisks[3] != disks[6] ||
+		orderedDisks[4] != disks[10] ||
+		orderedDisks[5] != disks[5] ||
+		orderedDisks[6] != disks[11] ||
+		orderedDisks[7] != disks[4] ||
+		orderedDisks[8] != disks[12] ||
+		orderedDisks[9] != disks[3] ||
+		orderedDisks[10] != disks[13] ||
+		orderedDisks[11] != disks[2] ||
+		orderedDisks[12] != disks[14] ||
+		orderedDisks[13] != disks[1] ||
+		orderedDisks[14] != disks[15] ||
+		orderedDisks[15] != disks[0] {
+		t.Errorf("getOrderedDisks returned incorrect order.")
+	}
+}
+
+// Test for isSuccessDataBlocks and isSuccessDecodeBlocks
+func TestIsSuccessBlocks(t *testing.T) {
+	dataBlocks := 8
+	parityBlocks := 8
+	enBlocks := make([][]byte, dataBlocks+parityBlocks)
+	for i := 0; i < dataBlocks+parityBlocks; i++ {
+		enBlocks[i] = []byte{'a'}
+	}
+	// When all data blocks are available.
+	if isSuccessDataBlocks(enBlocks, dataBlocks) == false {
+		t.Errorf("isSuccessDataBlocks should return true.")
+	}
+	enBlocks[0] = nil
+	// When one data block is not available.
+	if isSuccessDataBlocks(enBlocks, dataBlocks) == true {
+		t.Errorf("isSuccessDataBlocks should return false.")
+	}
+	if isSuccessDecodeBlocks(enBlocks, dataBlocks) == false {
+		t.Errorf("isSuccessDecodeBlocks should return true.")
+	}
+	for i := 1; i < dataBlocks-1; i++ {
+		enBlocks[i] = nil
+	}
+	// When one data block and all parity blocks are available.
+	if isSuccessDecodeBlocks(enBlocks, dataBlocks) == false {
+		t.Errorf("isSuccessDecodeBlocks should return true.")
+	}
+	enBlocks[dataBlocks-1] = nil
+	// When only parity blocks are available.
+	if isSuccessDecodeBlocks(enBlocks, dataBlocks) == true {
+		t.Errorf("isSuccessDecodeBlocks should return false.")
+	}
+}
+
+// Wrapper function for testGetReadDisks, testGetOrderedDisks.
+func TestErasureReadFile(t *testing.T) {
+	objLayer, dirs, err := getXLObjectLayer()
```

Comment:
can you use `makeTestBackend` function from `test-utils_test.go`, its a wrapper and used around in the tests/benchmarks.